.gitmodules

```diff
@@ -70,3 +70,7 @@
 [submodule "s3select"]
 	path = src/s3select
 	url = https://github.com/ceph/s3select.git
+[submodule "src/aws-sdk-cpp"]
+	path = src/aws-sdk-cpp
+	url = https://github.com/prateek2211/aws-sdk-cpp
```

Comment:
Do we want a more permanent/common upstream for this?

Reply:
you are right, we should have  https://github.com/ceph/aws-sdk-cpp
@mattbenjamin do you remember what was the process where we added s3select as a repo to the ceph org?

Reply:
@tchaikov could you please help here? 
We want to create an aws-sdk-cpp repo under the ceph org. This should be a fork of https://github.com/aws/aws-sdk-cpp.
We should add the changes we made in https://github.com/prateek2211/aws-sdk-cpp to the new repo, and then make it a submodule of ceph(pointing to the branch where we made the changes)

Reply:
i forked aws/aws-sdk-cpp in ceph/aws-sdk-cpp.

Reply:
> i forked aws/aws-sdk-cpp in ceph/aws-sdk-cpp.

@tchaikov thanks!

@prateek2211 can you please create a PR there with your changes from your fork?

Reply:
@yuvalif @prateek2211 i checked https://github.com/prateek2211/aws-sdk-cpp/commits/aws-ceph-submodule. the commit of https://github.com/prateek2211/aws-sdk-cpp/commit/f0a53b0e71d451c56daad5342dc6db3057e32c39 contains **lots* of changes 

> Showing 50,311 changed files with 86 additions and 8,233,272 deletions.

most of which are not related to what is claimed in the commit message

> Remove modules unusable for ceph and disable virtual overload function warning

@prateek2211 could you help elaborate on this?

Reply:
@tchaikov since we need only lambda and sns from the aws sdk (together with some "common" libs), we wanted to avoid adding huge amount of code into the ceph codebase (IIRC the aws sdk has more LoC than ceph...).
compilation time (and even cloning time) of the entire aws sdk is very long, and unnecessary.

Reply:
@yuvalif to fork upstream in this way is not a sustainable approach to address this problem. couple questions:

1. does aws-sdk-cpp offer finer grained cmake target so we can selectively build part of it, instead of build all of them in a wholesale?
1. regarding "adding huge amount of code into the ceph codebase", have you considered adding this project as an external project instead of including it as a submodule? so it's optional to our user?


FWIW, since it's included as a submodule, the full history is included anyway, i don't think the time for cloning this project changes much even if the latest commit removes the unused files.

Reply:
> @yuvalif to fork upstream in this way is not a sustainable approach to address this problem. for instance, how do we keep track of the upstream changes. how to do the rebase? 

had discussion over that in the stand-up meeting (thanks @alimaredia @cbodley @mkogan1):
* will investigate the option of building the necessary packages (for our CI and shaman targets) and storing them in shaman - similarly to what we did with boost
* however, this will take more time, and as we want to merge this PR, and minimize the impact on CI, would like to keep the trimmed down branch for now
* rebasing from upstream is expected only in case of severe bugs that we find (since we are using the SDK in a very specific way). I will own that

> also couple questions:
> 
> 1. does aws-sdk-cpp offer finer grained cmake target so we can selectively build part of it, instead of build all of them in a wholesale?

yes, we set the option to build only what we need. so build time is not too bad

> 2. regarding "adding huge amount of code into the ceph codebase", have you considered adding this project as an external project instead of including it as a submodule? so it's optional to our user?
> 

we are going to set the flag that include that to "ON" by default, so this won't help CI/shaman etc.

> FWIW, since it's included as a submodule, the full history is included anyway, and we don't do a shallow clone. i don't think the time for cloning this project changes much even if the latest commit removes the unused files.

I think we can configure [shallow clone](https://stackoverflow.com/questions/2144406/how-to-make-shallow-git-submodules) for that submodule